### ISSUE TITLE
Permission Errors between Version 1.1.16 and 1.1.15

### ISSUE THREAD

**Issue by rastacalavera** [borgbackup/borg#6100]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

Question/Issue

## System information. For client/server mode post info for both machines.
Atomic Pi x86 running Ubuntu Server 20.04.3 LTS
Raspberry Pi 4 Model B Rev 1.2 running Ubuntu Server 21.04 aarch64
Motile Laptop x86 running Zorin OS 16 (Ubuntu base from 20.04.2)

#### Your borg version (borg -V).
Atomic and motile have 1.1.15
RaspberryPi 1.1.16

#### Operating system (distribution) and version.
See above but all ubuntu based but mix between 20.4 and 21.04

#### Hardware / network configuration, and filesystems used.
all internal lan wired with ext4
All are using ssh keys and a non standard port
All machines can be ssh'd into each other with out issue.

#### How much data is handled by borg?
Less than 5 GB

#### Full borg commandline that lead to the problem (leave away excludes and passwords)
```
rastacalavera@pi4:~$ borg init --encryption=repokey ssh://atom@192.168.0.XXX:YYY//media/external/pi4
Enter new passphrase: 
Enter same passphrase again: 
Do you want your passphrase to be displayed for verification? [yN]: 
Local Exception
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 4690, in main
    exit_code = archiver.run(args)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 4622, in run
    return set_ec(func(args))
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 177, in wrapper
    return method(self, args, repository=repository, **kwargs)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 293, in do_init
    key = key_creator(repository, args)
  File "/usr/lib/python3/dist-packages/borg/crypto/key.py", line 110, in key_creator
    return key.create(repository, args)
  File "/usr/lib/python3/dist-packages/borg/crypto/key.py", line 688, in create
    key.init_ciphers()
  File "/usr/lib/python3/dist-packages/borg/crypto/key.py", line 419, in init_ciphers
    self.nonce_manager = NonceManager(self.repository, self.enc_cipher, manifest_nonce)
  File "/usr/lib/python3/dist-packages/borg/crypto/nonces.py", line 22, in __init__
    self.nonce_file = os.path.join(get_security_dir(self.repository.id_str), 'nonce')
  File "/usr/lib/python3/dist-packages/borg/helpers.py", line 546, in get_security_dir
    security_dir = os.environ.get('BORG_SECURITY_DIR', os.path.join(get_config_dir(), 'security'))
  File "/usr/lib/python3/dist-packages/borg/helpers.py", line 589, in get_config_dir
    os.makedirs(config_dir)
  File "/usr/lib/python3.9/os.py", line 225, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/home/rastacalavera/.config/borg'

Platform: Linux pi4 5.11.0-1015-raspi #16-Ubuntu SMP PREEMPT Wed Jul 14 11:04:58 UTC 2021 aarch64
Linux: Unknown Linux  
Borg: 1.1.16  Python: CPython 3.9.5 msgpack: 0.5.6.+borg1
PID: 1141421  CWD: /home/rastacalavera
sys.argv: ['/usr/bin/borg', 'init', '--encryption=repokey', 'ssh://atom@192.168.0.XXX:YYYY//media/external/pi4']
SSH_ORIGINAL_COMMAND: None

rastacalavera@pi4:~$ borg list ssh://atom@192.168.0.XXX:YYY//media/external/pi4
Repository has no manifest.
```

## Describe the problem you're observing.
When initializing and push/pulling between the laptop and atomic pi, there are never any issues. When trying to init from the raspberry pi4 to the atomic pi or laptop, there are permission and missing manifest issues.

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

Yes. One thing I am unsure about is when using the laptop and the atomic pi, I get this notification:

```
By default repositories initialized with this version will produce security
errors if written to with an older version (up to and including Borg 1.0.8).

If you want to use these older versions, you can disable the check by running:
borg upgrade --disable-tam 'ssh://atom@192.168.0.142:4140/~/141'

See https://borgbackup.readthedocs.io/en/stable/changes.html#pre-1-0-9-manifest-spoofing-vulnerability for details about the security implications.
```
Could this be related to my issues with the pi4 using 1.1.16? I don't think so but I may be wrong . . . 

#### Include any warning/errors/backtraces from the system logs

Please advise on how to do this and I can.

**Comment by ThomasWaldmann** [comment]
This is not borg version related, but an administrative issue. You should always use the same user to access borg files (the repo, the cache, etc.) - if you use sometimes root, sometimes non-root, you will get a mix of fs permissions and run into issues when running as non-root.

For a borg server, it is a good practice is to add a user `borg` and always use that user to access the repo files.

On the client side, you can either run as non-root (e.g. if you only back up files in your home dir) or as root (if you want to back up everything).

There might be also some infos in the docs about this. :-)

**Comment by rastacalavera** [comment]
Thanks for the reply. I read that part about the "borg" user but figured since I was using the user "atom" from both the laptop and the pi4 for the init and create, it was basically the same idea, just under a different name.
I made a "borg" user on the atomic pi which is acting as the server ( assuming the computer I init and create from is the client, the raspberry pi4 ) and got the same type of errors

```
rastacalavera@pi4:~$ borg init --encryption=repokey ssh://borg@192.168.0.142:4140//media/external/RaspberryPi
Enter new passphrase: 
Enter same passphrase again: 
Do you want your passphrase to be displayed for verification? [yN]: 
Local Exception
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 4690, in main
    exit_code = archiver.run(args)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 4622, in run
    return set_ec(func(args))
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 177, in wrapper
    return method(self, args, repository=repository, **kwargs)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 293, in do_init
    key = key_creator(repository, args)
  File "/usr/lib/python3/dist-packages/borg/crypto/key.py", line 110, in key_creator
    return key.create(repository, args)
  File "/usr/lib/python3/dist-packages/borg/crypto/key.py", line 688, in create
    key.init_ciphers()
  File "/usr/lib/python3/dist-packages/borg/crypto/key.py", line 419, in init_ciphers
    self.nonce_manager = NonceManager(self.repository, self.enc_cipher, manifest_nonce)
  File "/usr/lib/python3/dist-packages/borg/crypto/nonces.py", line 22, in __init__
    self.nonce_file = os.path.join(get_security_dir(self.repository.id_str), 'nonce')
  File "/usr/lib/python3/dist-packages/borg/helpers.py", line 546, in get_security_dir
    security_dir = os.environ.get('BORG_SECURITY_DIR', os.path.join(get_config_dir(), 'security'))
  File "/usr/lib/python3/dist-packages/borg/helpers.py", line 589, in get_config_dir
    os.makedirs(config_dir)
  File "/usr/lib/python3.9/os.py", line 225, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/home/rastacalavera/.config/borg'

Platform: Linux pi4 5.11.0-1015-raspi #16-Ubuntu SMP PREEMPT Wed Jul 14 11:04:58 UTC 2021 aarch64
Linux: Unknown Linux  
Borg: 1.1.16  Python: CPython 3.9.5 msgpack: 0.5.6.+borg1
PID: 1180626  CWD: /home/rastacalavera
sys.argv: ['/usr/bin/borg', 'init', '--encryption=repokey', 'ssh://borg@192.168.0.142:4140//media/external/RaspberryPi']
SSH_ORIGINAL_COMMAND: None
``` 
So where am I getting lost here? Do I need "borg" users on the client and server? would I use the borg user to back up my rastacalavera files? that doesn't seem right . . . 

I can do borg init and creates and restores and deletes just fine using the "atom" user from my laptop (motile) it's just from the pi4 that is the issue.

**Comment by ThomasWaldmann** [comment]
To see the issue, just do:
```
ls -l /home/rastacalavera/.config/borg
```

Then use this to fix the issue:
```
chown -R rastacalavera.rastacalavera /home/rastacalavera/.config/borg
```

**Comment by rastacalavera** [comment]
On the pi4, I ran ` ls -l /home/rastacalavera/.config/borg` and got:
```
rastacalavera@pi4:~$ ls -l /home/rastacalavera/.config/borg
ls: cannot access '/home/rastacalavera/.config/borg': No such file or directory
```
When I look in the `.config` folder, there is only a folder called "appdata" which is empty.  That folder exists on the other two machines. I completely purged my borg repos and the borg folder in `.config` on the laptop and atomic pi to see if when I reinstall borg, that folder is created. My assumption was correct but when reinstalled on the raspberry Pi 4, that folder is not created upon installation. 

I looked at the `.config` folder permissions and it turns out that it was `root:root` instead of `rastacalavera:rastacalavera`. I did `chown -R rastacalavera:rastacalavera ~/.config/` then tried to create a local repo and it worked without issue. I then tried to do the remote repo and it worked without issue.

Very odd mistake on my part. No idea how that folder was taken over by root . . .  thank you for your help in resolving this and helping me find a solution.
